PdCore:pd-core:javadoc task reports 98 errors

```
> Task :PdCore:pd-core:javadoc
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/utils/Properties.java:10: error: package android.os does not exist
import android.os.Build;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/utils/PdUiDispatcher.java:12: error: package android.os does not exist
import android.os.Handler;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/utils/PdUiDispatcher.java:13: error: package android.util does not exist
import android.util.Log;
                   ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/utils/PdUiDispatcher.java:27: error: cannot find symbol
	private final Handler handler;
	              ^
  symbol:   class Handler
  location: class PdUiDispatcher
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/midi/MidiToPdAdapter.java:11: error: package android.media.midi does not exist
import android.media.midi.MidiReceiver;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/midi/MidiToPdAdapter.java:19: error: cannot find symbol
public class MidiToPdAdapter extends MidiReceiver {
                                     ^
  symbol: class MidiReceiver
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/midi/PdToMidiAdapter.java:11: error: package android.media.midi does not exist
import android.media.midi.MidiInputPort;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/midi/PdToMidiAdapter.java:23: error: cannot find symbol
	private Map<Integer, MidiInputPort> inputPorts = new HashMap<Integer, MidiInputPort>();
	                     ^
  symbol:   class MidiInputPort
  location: class PdToMidiAdapter
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/midi/PdToMidiAdapter.java:54: error: cannot find symbol
	public void open(MidiInputPort inputPort, int pdPort) {
	                 ^
  symbol:   class MidiInputPort
  location: class PdToMidiAdapter
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/midi/PdToMidiAdapter.java:64: error: cannot find symbol
	public void close(MidiInputPort inputPort) {
	                  ^
  symbol:   class MidiInputPort
  location: class PdToMidiAdapter
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:15: error: package android.app does not exist
import android.app.Notification;
                  ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:16: error: package android.app does not exist
import android.app.NotificationChannel;
                  ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:17: error: package android.app does not exist
import android.app.NotificationManager;
                  ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:18: error: package android.app does not exist
import android.app.PendingIntent;
                  ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:19: error: package android.app does not exist
import android.app.Service;
                  ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:20: error: package android.content does not exist
import android.content.Intent;
                      ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:21: error: package android.content does not exist
import android.content.SharedPreferences;
                      ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:22: error: package android.content.pm does not exist
import android.content.pm.ServiceInfo;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:23: error: package android.content.res does not exist
import android.content.res.Resources;
                          ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:24: error: package android.os does not exist
import android.os.Binder;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:25: error: package android.os does not exist
import android.os.Build;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:26: error: package android.os does not exist
import android.os.IBinder;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:27: error: package android.preference does not exist
import android.preference.PreferenceManager;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:28: error: package android.util does not exist
import android.util.Log;
                   ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:40: error: cannot find symbol
public class PdService extends Service {
                               ^
  symbol: class Service
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:42: error: cannot find symbol
	public class PdBinder extends Binder {
	                              ^
  symbol:   class Binder
  location: class PdService
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:177: error: cannot find symbol
	public synchronized void startAudio(Intent intent, int icon, String title, String description) {
	                                    ^
  symbol:   class Intent
  location: class PdService
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:186: error: cannot find symbol
	public synchronized void startAudio(Notification notification) {
	                                    ^
  symbol:   class Notification
  location: class PdService
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:216: error: cannot find symbol
	public IBinder onBind(Intent intent) {
	                      ^
  symbol:   class Intent
  location: class PdService
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:216: error: cannot find symbol
	public IBinder onBind(Intent intent) {
	       ^
  symbol:   class IBinder
  location: class PdService
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:221: error: cannot find symbol
	public boolean onUnbind(Intent intent) {
	                        ^
  symbol:   class Intent
  location: class PdService
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:250: error: cannot find symbol
	private Notification makeNotification(Intent intent, int icon, String title, String description) {
	                                      ^
  symbol:   class Intent
  location: class PdService
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:250: error: cannot find symbol
	private Notification makeNotification(Intent intent, int icon, String title, String description) {
	        ^
  symbol:   class Notification
  location: class PdService
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdService.java:271: error: cannot find symbol
	private void startForeground(Notification notification) {
	                             ^
  symbol:   class Notification
  location: class PdService
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:14: error: package android.content does not exist
import android.content.Context;
                      ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:15: error: package android.content does not exist
import android.content.SharedPreferences;
                      ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:16: error: package android.content.res does not exist
import android.content.res.Resources;
                          ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:17: error: package android.os does not exist
import android.os.Bundle;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:18: error: package android.preference does not exist
import android.preference.PreferenceActivity;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:19: error: package android.preference does not exist
import android.preference.PreferenceManager;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:20: error: package android.preference does not exist
import android.preference.PreferenceFragment;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:30: error: cannot find symbol
public class PdPreferences extends PreferenceActivity {
                                   ^
  symbol: class PreferenceActivity
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:9: error: cannot find symbol
import org.puredata.android.service.R;
                                   ^
  symbol:   class R
  location: package org.puredata.android.service
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:11: error: package android.content does not exist
import android.content.Context;
                      ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:12: error: package android.app does not exist
import android.app.Activity;
                  ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:13: error: package android.media does not exist
import android.media.AudioManager;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:14: error: package android.media does not exist
import android.media.AudioDeviceCallback;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:15: error: package android.media does not exist
import android.media.AudioDeviceInfo;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:16: error: package android.util does not exist
import android.util.Log;
                   ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:17: error: package android.content.res does not exist
import android.content.res.Resources;
                          ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:18: error: package android.preference does not exist
import android.preference.ListPreference;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:19: error: package android.preference does not exist
import android.preference.PreferenceFragment;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:35: error: cannot find symbol
	protected void onCreate(Bundle savedInstanceState) {
	                        ^
  symbol:   class Bundle
  location: class PdPreferences
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:64: error: cannot find symbol
	public static void initPreferences(Context context) {
	                                   ^
  symbol:   class Context
  location: class PdPreferences
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:27: error: cannot find symbol
	private AudioManager mAudioManager;
	        ^
  symbol:   class AudioManager
  location: class AudioDevices
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:59: error: cannot find symbol
	public AudioDevices(Activity context) {
	                    ^
  symbol:   class Activity
  location: class AudioDevices
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:30: error: cannot find symbol
		private ListPreference mPref;
		        ^
  symbol:   class ListPreference
  location: class AudioDevices.AudioDeviceList
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:39: error: cannot find symbol
		AudioDeviceList(PreferenceFragment prefFragment, String key) {
		                ^
  symbol:   class PreferenceFragment
  location: class AudioDevices.AudioDeviceList
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java:44: error: cannot find symbol
		public void add(AudioDeviceInfo device) {
		                ^
  symbol:   class AudioDeviceInfo
  location: class AudioDevices.AudioDeviceList
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:47: error: cannot find symbol
	public static class MyPreferenceFragment extends PreferenceFragment
	                                                 ^
  symbol:   class PreferenceFragment
  location: class PdPreferences
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/service/PdPreferences.java:50: error: cannot find symbol
		public void onCreate(final Bundle savedInstanceState)
		                           ^
  symbol:   class Bundle
  location: class MyPreferenceFragment
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioRecordWrapper.java:14: error: package android.media does not exist
import android.media.AudioFormat;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioRecordWrapper.java:15: error: package android.media does not exist
import android.media.AudioRecord;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioRecordWrapper.java:16: error: package android.media does not exist
import android.media.MediaRecorder;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioRecordWrapper.java:17: error: package android.os does not exist
import android.os.Process;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioRecordWrapper.java:31: error: cannot find symbol
	private AudioRecord rec = null;
	        ^
  symbol:   class AudioRecord
  location: class AudioRecordWrapper
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioFormatUtil.java:12: error: package android.media does not exist
import android.media.AudioFormat;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioFormatUtil.java:13: error: package android.util does not exist
import android.util.Log;
                   ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/PdAudio.java:15: error: package android.content does not exist
import android.content.Context;
                      ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/PdAudio.java:16: error: package android.os does not exist
import android.os.Handler;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/PdAudio.java:17: error: package android.os does not exist
import android.os.Looper;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:15: error: package android.annotation does not exist
import android.annotation.TargetApi;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:16: error: package android.content does not exist
import android.content.Context;
                      ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:17: error: package android.media does not exist
import android.media.AudioFormat;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:18: error: package android.media does not exist
import android.media.AudioManager;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:19: error: package android.media does not exist
import android.media.AudioRecord;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:20: error: package android.media does not exist
import android.media.AudioTrack;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:21: error: package android.media does not exist
import android.media.MediaPlayer;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:22: error: package android.os does not exist
import android.os.Build;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:23: error: package android.os does not exist
import android.os.Process;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:24: error: package android.util does not exist
import android.util.Log;
                   ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/PdAudio.java:31: error: cannot find symbol
	private static final Handler handler = new Handler(Looper.getMainLooper());
	                     ^
  symbol:   class Handler
  location: class PdAudio
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/PdAudio.java:98: error: cannot find symbol
	public synchronized static void startAudio(Context context) {
	                                           ^
  symbol:   class Context
  location: class PdAudio
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:40: error: cannot find symbol
	private final AudioTrack track;
	              ^
  symbol:   class AudioTrack
  location: class AudioWrapper
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:93: error: cannot find symbol
	public synchronized void start(Context context) {
	                               ^
  symbol:   class Context
  location: class AudioWrapper
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioWrapper.java:166: error: cannot find symbol
	private void avoidClickHack(Context context) {
	                            ^
  symbol:   class Context
  location: class AudioWrapper
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:12: error: package android.annotation does not exist
import android.annotation.TargetApi;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:13: error: package android.content does not exist
import android.content.Context;
                      ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:14: error: package android.content.pm does not exist
import android.content.pm.PackageManager;
                         ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:15: error: package android.media does not exist
import android.media.AudioFormat;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:16: error: package android.media does not exist
import android.media.AudioManager;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:17: error: package android.media does not exist
import android.media.AudioRecord;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:18: error: package android.media does not exist
import android.media.AudioTrack;
                    ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:19: error: package android.os does not exist
import android.os.Build;
                 ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:20: error: package android.util does not exist
import android.util.Log;
                   ^
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:42: error: cannot find symbol
	public static synchronized void init(Context context) {
	                                     ^
  symbol:   class Context
  location: class AudioParameters
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:254: error: cannot find symbol
	@TargetApi(17)  // Using lazy class loading trick to hide new features from old devices.
	 ^
  symbol:   class TargetApi
  location: class AudioParameters
/home/runner/work/pd-for-android/pd-for-android/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioParameters.java:265: error: cannot find symbol
		static JellyBeanMR1OpenSLParameters getParameters(Context context) {
		                                                  ^
  symbol:   class Context
  location: class JellyBeanMR1OpenSLParameters
98 errors
```

--- a/PdCore/pd-core/build.gradle
+++ b/PdCore/pd-core/build.gradle
@@ -96,8 +96,8 @@ task sourcesJar(type: Jar) {
 
 task javadoc(type: Javadoc) {
     source = android.sourceSets.main.java.srcDirs
-    classpath += files("${android.sdkDirectory}/platforms/${android.compileSdk}/android.jar")
-    failOnError = false // TODO: fix javadoc issues
+    classpath += files("${android.sdkDirectory}/platforms/android-${android.compileSdk}/android.jar")
+    failOnError = true
 }
 
 task javadocJar(type: Jar, dependsOn: javadoc) {

--- a/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java
+++ b/PdCore/pd-core/src/main/java/org/puredata/android/io/AudioDevices.java
@@ -6,7 +6,6 @@
  */
 
 package org.puredata.android.io;
-import org.puredata.android.service.R;
 
 import android.content.Context;
 import android.app.Activity;
@@ -60,8 +59,10 @@ public class AudioDevices {
 		mAudioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
 		PreferenceFragment prefFragment = (PreferenceFragment) context.getFragmentManager().findFragmentByTag("prefFragment");
 		Resources res = context.getResources();
-		mInputDevices = new AudioDeviceList(prefFragment, res.getString(R.string.pref_key_indevice));
-		mOutputDevices = new AudioDeviceList(prefFragment, res.getString(R.string.pref_key_outdevice));
+		final String inputKey = res.getString(org.puredata.android.service.R.string.pref_key_indevice);
+		mInputDevices = new AudioDeviceList(prefFragment, inputKey);
+		final String outputKey = res.getString(org.puredata.android.service.R.string.pref_key_outdevice);
+		mOutputDevices = new AudioDeviceList(prefFragment, outputKey);
 		setupAudioDeviceCallback();
 	}
 

